packages/api/src/api/ProviderApi.ts

```diff
@@ -30,18 +32,30 @@ export default class ProviderApi extends HttpClientBase {
         return this.axios.get(`/v1/prosopo/providers`)
     }
 
-    public getCaptchaChallenge(
+    public async getCaptchaChallenge(
         userAccount: string,
-        randomProvider: ProsopoRandomProvider
+        randomProvider: ProsopoRandomProvider,
+        dappAccount: string,
+        retryAttempt = 0
     ): Promise<GetCaptchaResponse> {
         const { provider } = randomProvider
         const { blockNumber } = randomProvider
 
-        return this.axios.get(
-            `/v1/prosopo/provider/captcha/${provider.datasetId}/${userAccount}/${
-                this.network.dappContract.address
-            }/${blockNumber.toString().replace(/,/g, '')}`
-        )
+        if (retryAttempt > RETRY_LIMIT) {
+            throw new Error('Failed to get captcha challenge after maximum retry attempts')
+        }
+
+        try {
+            return this.axios.get(
+                `/v1/prosopo/provider/captcha/${provider.datasetId}/${userAccount}/${
+                    this.network.dappContract.address
+                }/${blockNumber.toString().replace(/,/g, '')}`
+            )
+        } catch (e) {
+            console.log(e)
+            const newRandomProvider = await this.getNewRandomProvider(userAccount)
+            return this.getCaptchaChallenge(userAccount, newRandomProvider, dappAccount, retryAttempt + 1)
```

Comment:
you're making this function recursive here. I think it would be better to do this in a loop rather than having the retry parameter. E.g.
```
for retry in 0..10 {
  challenge = getChallenge()
  if challenge then return
}
throw new Error("max retries exceeded")
```

Reply:
Good shout, this avoids global consts :)